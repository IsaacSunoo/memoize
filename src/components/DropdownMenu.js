import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class DropdownMenu extends Component {
    render() {
        return (
            <div className='menu-dropdown'>
                <ul>
                    <li><Link className='dropdown-link' onClick={() => { this.props.displayDropdown() }} to='/'>Home</Link></li>
                    <li><Link className='dropdown-link' onClick={() => { this.props.displayDropdown() }} to='/flashCards'>FlashCards</Link></li>
                    <li><Link className='dropdown-link' onClick={() => { this.props.displayDropdown() }} to='/quiz'>Prototypes Quiz</Link></li>
                    <li><Link className='dropdown-link' onClick={() => { this.props.displayDropdown() }} to='/results'>Quiz Results</Link></li>
                    <li><Link className='dropdown-link' onClick={() => { this.props.displayDropdown() }} to='/coding'>Practice Coding</Link></li>
                </ul>
            </div>
        )
    }
}

export default DropdownMenu;