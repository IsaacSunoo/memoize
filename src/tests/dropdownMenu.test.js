import React from 'react';
import DropdownMenu from '../components/DropdownMenu';
import { shallow } from 'enzyme';

describe('DropdownMenu', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <DropdownMenu />
        )
    })

    it('should match the snapshot with all data passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

