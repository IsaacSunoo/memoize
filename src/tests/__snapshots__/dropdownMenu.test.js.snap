// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu should match the snapshot with all data passed in correctly 1`] = `
<div
  className="menu-dropdown"
>
  <ul>
    <li>
      <Link
        className="dropdown-link"
        replace={false}
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        className="dropdown-link"
        replace={false}
        to="/flashCards"
      >
        FlashCards
      </Link>
    </li>
    <li>
      <Link
        className="dropdown-link"
        replace={false}
        to="/quiz"
      >
        Prototypes Quiz
      </Link>
    </li>
    <li>
      <Link
        className="dropdown-link"
        replace={false}
        to="/coding"
      >
        Practice Coding
      </Link>
    </li>
  </ul>
</div>
`;
